test/parallel/test-zlib-failed-init.js

```diff
@@ -6,11 +6,15 @@ const assert = require('assert');
 const zlib = require('zlib');
 
 // For raw deflate encoding, requests for 256-byte windows are rejected as
-// invalid by zlib.
-// (http://zlib.net/manual.html#Advanced)
-assert.throws(() => {
-  zlib.createDeflateRaw({ windowBits: 8 });
-}, /^Error: Init error$/);
+// invalid by zlib (http://zlib.net/manual.html#Advanced).
+// This check was introduced in vesion 1.2.9 and prior to that there was
+// no such rejection which is the reason for the version check below
+// (http://zlib.net/ChangeLog.txt).
+if (/^\d+\.\d+\.[9]|\d{2,}$/.test(process.versions.zlib)) {
```

Comment:
It could probably be made stricter, yes. I also don’t quit get the `|\d{2,}$` part – are there missing parentheses? `/^a|b$/` is grouped as `/(^a)|(b$)/`, not `/^(a|b)$/`, but you probably meant something like the latter?